hal/src/timer.rs

```diff
@@ -47,11 +47,26 @@ where
         T: Into<Hertz>,
     {
         let timeout = timeout.into();
-        let params = clock::ClockParams::new(self.freq, timeout);
-        let divider = params.divider;
+        let ticks: u32 = self.freq.0/timeout.0.saturating_sub(1).max(1);
+        let overflow: u16 = ((ticks >> 16) + 1) as u16;
```

Comment:
Can we fix this in `clock::ClockParams::new` instead of duplicating most of its code out into two other places?

Reply:
ClockParams::new is supposed to find the nearest matching frequency, so it's not broken. The usage of it here is and I can't think of a nice way of nudging it into what we want it to do. (Find the highest possible frequency, that's less than 2^16 * desired frequency)

Reply:
I wasn't sure of all the places it was used and didn't want to cause any side effects elsewhere, but if you think it's fine I'll go ahead with it.

Reply:
@sajattack This also doesn't **need** to be u16, so just remove it & it's probably nicer to just combine it with the next line

Reply:
I think this stuff is **really** timer specific & it's duplicated either way, since we also have 2 clock.rs files. Since the `ClockParams::new` isn't used anywhere else, we can probably just delete it.

May be nice to do a separate timer method for determining the timing, since the `new` method is getting pretty hairy